hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -198,7 +198,7 @@ private void inlineCompaction(HoodieTable table, Option<Map<String, String>> ext
    * @return Collection of Write Status
    */
   protected HoodieWriteMetadata<O> logCompact(String logCompactionInstantTime, boolean shouldComplete) {
-    HoodieTable<?, I, ?, T> table = createTable(config, context.getHadoopConf().get());
+    HoodieTable<?, I, ?, T> table = createTable(config, (Configuration) context.getStorageConf().unwrap());
```

Comment:
We can add a interface like `#unwrapAs(Class<T> clazz)` to unwrap the class as specific type to eliminate the type cast.

Reply:
New util methods are added in this one so they are clear: #11121